src/common/snippets/src/runtime_configurator.cpp

```diff
@@ -272,10 +266,187 @@ void RuntimeConfigurator::update_data_offsets() const {
     }
 }
 
-void RuntimeConfigurator::update_latest_shapes() {
-    for (size_t i = 0; i < m_io_num; ++i) {
-        m_latest_shapes[i] = m_io_descs[i]->get_shape();
+std::vector<VectorDims> RuntimeConfigurator::extract_shapes() const {
+    std::vector<VectorDims> shapes(m_io_num);
+    for (size_t i = 0; i < m_io_num; ++i)
+        shapes[i] = m_io_descs[i]->get_shape();
+    return shapes;
+}
+
+std::vector<std::vector<size_t>> RuntimeConfigurator::extract_layouts() const {
+    std::vector<std::vector<size_t>> layouts(m_io_num);
+    for (size_t i = 0; i < m_io_num; ++i)
+        layouts[i] = m_io_descs[i]->get_layout();
+    return layouts;
+}
+
+RuntimeConfigurator::UnifiedLoopInfoRtParams RuntimeConfigurator::compute_runtime_params(const lowered::UnifiedLoopInfoPtr& loop_info) {
+    RuntimeConfigurator::UnifiedLoopInfoRtParams rt_params;
+    rt_params.work_amount = loop_info->get_work_amount();
+    const auto count = loop_info->get_input_count() + loop_info->get_output_count();
+    rt_params.ptr_increments.resize(count);
+    rt_params.finalization_offsets.resize(count);
+
+    size_t idx = 0;
+    loop_info->iterate_through_descs(
+        [&rt_params, &idx](const lowered::UnifiedLoopInfo::LoopPortDesc& desc) {
+            rt_params.ptr_increments[idx] = desc.ptr_increment;
+            rt_params.finalization_offsets[idx] = desc.finalization_offset;
+            ++idx;
+        });
+    return rt_params;
+}
+
+void RuntimeConfigurator::ParallelWAOptimizer::init(const lowered::LinearIRPtr& linear_ir,
+                                                    const std::vector<PortDescriptorPtr>& io_descs,
+                                                    size_t in_num) {
+    if (linear_ir->get_config().m_enable_domain_optimization || !linear_ir->is_dynamic())
+        return;
+
+    const auto brgemms = find_applicable_brgemms(linear_ir);
+    // Parallel WA optimization is Brgemm related
+    if (brgemms.empty())
+        return;
+
+    concurrency = linear_ir->get_config().m_min_parallel_work_amount;
+    unsqueezed_params = find_unsqueezed_params(linear_ir, brgemms);
+    OPENVINO_ASSERT(!unsqueezed_params.empty(), "unsqueezed_params mustn't be empty after initialization");
```

Comment:
Could you please add the comment with brief description?
I left the similar comment here: https://github.com/openvinotoolkit/openvino/pull/25733#discussion_r1704928983

Reply:
Done. thanks